### DISCUSSION TITLE
2D Segmentation

### DISCUSSION THREAD

**Discussion by tk6684** [Project-MONAI/tutorials#1186] [General]
I am trying to develop segmentation for CT scan of head. I have my own data for images and labels/masks in two separate folders. An example of data that I have is attached. The 2D segmentation tutorial is based on synthetic data (https://github.com/Project-MONAI/tutorials/blob/main/2d_segmentation/torch/unet_training_array.py). How do I modify the code to fit my data? Thanks!

![Image1](https://user-images.githubusercontent.com/123214371/213824114-898c69a8-a4ce-4647-90a8-19d5923d7030.jpg)
![Labelledimage1](https://user-images.githubusercontent.com/123214371/213824121-887ecea8-d9ef-48c5-b4e6-81c2e4202ab3.png)

**Comment by KumoLiu** [comment]
Hi @tk6684, you could replace the part about generating synthetic data with the example I showed below(need to change the data path).
https://github.com/Project-MONAI/tutorials/blob/56ba738d2c05b55f1b4e9c2823210645aecfb5d0/2d_segmentation/torch/unet_training_array.py#L43-L50

```
# "img*.png" means all the image files start with 'img' and end with 'png'. You could update it based on your data.
images = sorted(glob(os.path.join(images_root_dir, "img*.png")))
# "seg*.png" means all the label files start with 'seg' and end with 'png'. 
segs = sorted(glob(os.path.join(segs_root_dir, "seg*.png")))
```

Hope it can help you, thanks!

**Reply by tk6684** [reply]
Thanks for the suggestion @KumoLiu . I am however having issue with the path itself. When I put the path for example- r'"C:\Users\tapan\OneDrive\Desktop\2D img", I am always getting an error that file not recognized or no permission.

If you want to collaborate on the project, I will be happy to. You can reach me at tapan.kavi@gmail.com

**Reply by KumoLiu** [reply]
Hi @tk6684, there appears to be a single egg quotation in the path you provided; perhaps this is the root of the problem. Alternatively, if you share the entire error message, maybe I could help find more possible reasons. Thanks!

**Reply by tk6684** [reply]
@KumoLiu 
this is the error I get- 
AttributeError: 'str' object has no attribute 'read'
If this is the path to the folder on my laptop- C:\Users\tapan\OneDrive\Desktop\2D img, how would you put it in the code you provided above. Thanks!

**Reply by KumoLiu** [reply]
Something like this?
```
images_root_dir = r"C:\Users\tapan\OneDrive\Desktop\2D img"
images = sorted(glob(os.path.join(images_root_dir, "img*.png")))
```
